tests/test_sequences_and_iterators.py

```diff
@@ -72,13 +72,13 @@ def test_iterator_referencing():
     vec = m.VectorNonCopyableIntPair()
     vec.append([3, 4])
     vec.append([5, 7])
-    assert [int(x) for x in vec.keys()] == [3, 5]
```

Comment:
This check is kind of often wrong. I've had it trigger incorrectly in another package too. Not all objects play nice and have `iter(x) == iter(x.keys())`. I wish it was type-aware.